docs/optimizations.md

```diff
@@ -0,0 +1,28 @@
+# Optimizations and best practices
+Here are some of the best practices we apply to the `autonolas-governance` in order to make the code more gas optimized.
+Please note that the following statements concern our code only and do not account for the inherited non-modified code.
+
+- No modifiers;
+- No asserts or custom `Panic()` errors;
+- Custom `reverts` and no `require` statements;
+- Storage vs memory keyword: decision is on what to use after testing for gas spending;
+- Explicit type casting even where the compiler does it by default;
+- No usage of `safeTransfer()` from the OpenZeppelin (or their analogues) where it is not absolutely necessary;
+- Follow `Checks-Effects-Interactions` pattern;
+- Events are not subject to reentrancy attacks since events do not cause a function to throw;
+- Avoid reverts where the logic allows not to revert and return a zero value instead;
+- All `unchecked` statements are commented on;
+- All functions with `owner` privileges are not considered to be called from malicious actors and are not meant to be executed for other than common sense scenarios;
+- Our `OLAS` token is considered to be called `with good intentions`, and strictly corresponds to the ERC20 standard;
```

Comment:
What does "with good intentions" mean? Not sure what you're trying to say here.

Reply:
It means that the token code does not do contain any special harmful code. Will rephrase